test/src/test/java/org/corfudb/integration/StreamingIT.java

```diff
@@ -610,6 +631,113 @@ public void testSubscriberOnEmptyTables() throws Exception {
         assertThat(shutdownCorfuServer(corfuServer)).isTrue();
     }
 
+    /**
+     * Test the case where a subscriber hits an error (for instance due to a poison pill) and
+     * decides to re-subscribe from the last observed update (based on the provided timestamp).
+     */
+    @Test
+    @SuppressWarnings("checkstyle:magicnumber")
+    public void testSubscriberTimestamp() throws Exception {
+        // Run a corfu server.
+        Process corfuServer = runSinglePersistentServer(corfuSingleNodeHost, corfuStringNodePort);
+
+        // Start a Corfu runtime.
+        runtime = createRuntime(singleNodeEndpoint);
+        CorfuStore store = new CorfuStore(runtime);
+
+        // Record the initial timestamp.
+        Timestamp ts1 = store.getTimestamp();
+        final String namespace = "test_namespace";
+        final String tableNameA = "table_testA";
+        final String tableNameB = "table_testB";
+
+        // Create 2 tables.
+        Table<Uuid, SampleTableAMsg, Uuid> tableA = store.openTable(
+                namespace, tableNameA,
+                Uuid.class, SampleTableAMsg.class, Uuid.class,
+                TableOptions.builder().build()
+        );
+
+        Table<Uuid, SampleTableBMsg, Uuid> tableB = store.openTable(
+                namespace, tableNameB,
+                Uuid.class, SampleTableBMsg.class, Uuid.class,
+                TableOptions.builder().build()
+        );
+
+        final int numUpdates = 10;
+
+        // Subscribe to streaming for one 'tag' corresponding to TableA
+        // This listener will artificially throw an Exception when numUpdates/2 are received, enforcing unsubscription
+        // and testing the last provided timestamp truly tracks the last received update and can resume from that point
+        // onwards (no data loss or need to sync a full snapshot)
+        CountDownLatch latch = new CountDownLatch(1);
+        StreamListenerImpl listener1 = new StreamListenerImpl("stream_listener_1", numUpdates/2, latch);
+        store.subscribeListener(listener1, namespace, "sample_streamer_1",
+                Collections.singletonList(tableNameA), ts1);
+
```

Comment:
This timestamp is not reliable. In the case of a sequence regression, some deltas can be missed.

Reply:
not really, right? Cause this timestamp is obtained from the streaming API which is only called after a write is committed, so a sequencer regression cannot regress below this materialized point.

Reply:
`ts1` is just a sequencer call and it's not gaurenteed to be materialized `Timestamp ts1 = store.getTimestamp();` will call `runtime.getSequencerView().query().getToken();`

Reply:
I thought you were referring to the second one which comes from the LogData, that's the one I'm saying can't regress as it was captured on the onNext, when reading an update which is already persisted in the datastore.

This one, yes I agree, in this case sequence regression is a possibility which would incur in data loss. But this was designed earlier not part of this change... And actually this is what clients are doing, so probably we'll need a solution to this as you're pointing out, but that can be part of another PR.